addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1099 +1,1042 @@
+                // here label is an array of length 1 and contains a number
+                return this.dateClasses.representative(label, originIndex) || '';
+            } else if (this.propsCopy.mode === 'pie' && index === 0) {
+                // here label is an array of length at least one containing string or numbers
+                const labelCopy = label.slice();
+                let newLabel;
+                if (originIndex === undefined) {
+                    newLabel = this.dateClasses.dateClassMembers(label[index]);
+                } else {
+                    newLabel = this.dateClasses.representative(label[index], originIndex);
+                }
+                labelCopy.splice(index, 1, newLabel);
+                return labelCopy.join('/');
+            }
+            // here label is an array containing strings or numbers.
+            return label.join('/') || this.env._t("Total");
         }
-        this.$el.empty();
-        if (!_.contains(CHART_TYPES, this.state.mode)) {
-            this.trigger_up('warning', {
-                title: _t('Invalid mode for chart'),
-                message: _t('Cannot render chart with mode : ') + this.state.mode
+
+        /**
+         * Instantiate a Chart (Chart.js lib) to render the graph according to
+         * the current config. Chart.js performs the rendering in a
+         * nextAnimationFrame, so we wait for a tick before destroying/
+         * instantiating the chart, so that both are done in the same frame,
+         * and there is no flickering. Indeed, nested nextAnimationFrame
+         * callbacks are executed in successive frames, and when we reach this
+         * function, we already are in a requestAnimationFrame callback (see
+         * Owl internals).
+         * @private
+         * @returns {Promise}
+         */
+        async _renderGraphMainContent() {
+            await new Promise(setTimeout);
+
+            if (this.chart) {
+                this.chart.destroy();
+            }
+            if (this.children) {
+                for (const child of this.children) {
+                    child.remove();
+                }
+                delete this.children;
+            }
+
+            if (!this.el) {
+                // Element could have been destroyed in the interval. Rendering
+                // is aborted in that case.
+
+                // Is it normal?!
+                return;
+            }
+
+            const div = this._renderTemplate("web.GraphRenderer.MainContent", this);
+            this.children = [...div.children]
+            for (const child of this.children) {
+                this.el.append(child);
+            }
+
+            if (!this.noContentHelperData) {
+                const canvas = this.el.querySelector("canvas");
+                const ctx = canvas.getContext('2d');
+                this.chart = new Chart(ctx, this.config);
+            }
+
+            this._suppressFocusableElements();
```

Comment:
we have to wait this.el to be 'ready' in order to call this (cannot be done in mounted (super))